base/MAPL_sun_uc.F90

```diff
@@ -759,6 +807,160 @@ end function MAPL_SunOrbitCreate
 
 !==========================================================================
 
+!BOPI
+
+! !IROUTINE:  MAPL_SunOrbitCreateFromConfig
+
+! !DESCRIPTION:
+
+! Like MAPL_SunOrbitCreate() but gets orbital parameters from Config CF.
+
+! !INTERFACE:
+
+   function MAPL_SunOrbitCreateFromConfig ( &
+      CF, CLOCK, FIX_SUN, RC) result (ORBIT)
+
+! !ARGUMENTS:
+
+      type (ESMF_Config), intent(INOUT) :: CF
+      type (ESMF_Clock),  intent(IN   ) :: CLOCK
+      logical,            intent(IN   ) :: FIX_SUN
+      integer, optional,  intent(OUT  ) :: RC
+
+      type (MAPL_SunOrbit)              :: ORBIT
+
+!EOPI
+
+      character(len=ESMF_MAXSTR), parameter :: IAm = "SunOrbitCreateFromConfig"
+      integer :: STATUS
+
+      real :: ECC, OB, PER
+      integer :: EQNX
+
+      logical :: EOT, ORBIT_ANAL2B
+      integer :: ORB2B_REF_YYYYMMDD, ORB2B_REF_HHMMSS, &
+           ORB2B_EQUINOX_YYYYMMDD, ORB2B_EQUINOX_HHMMSS
+      real :: ORB2B_YEARLEN, &
+           ORB2B_ECC_REF, ORB2B_ECC_RATE, &
+           ORB2B_OBQ_REF, ORB2B_OBQ_RATE, &
+           ORB2B_LAMBDAP_REF, ORB2B_LAMBDAP_RATE
+
+      ! pmn: There is one orbit is per STATE, so, for example, the MAPL states of the
+      ! solar and land gridded components can potentially have independent solar orbits.
+      ! Usually these "independent orbits" will be IDENTICAL because the configuration
+      ! resources such as "ECCENTRICITY:" or "EOT:" will not be qualified by the name
+      ! of the gridded component. But for example, if the resource file specifies
+      !   "EOT: .FALSE."
+      ! but
+      !   "SOLAR_EOT: .TRUE."
+      ! then only SOLAR will have an EOT correction. The same goes for the new orbital
+      ! system choice ORBIT_ANAL2B.
+      !   A state's orbit is actually created in this routine by requesting the ORBIT
+      ! object. If its not already created then it will be made below. GridComps that
+      ! don't needed an orbit and dont request one will not have one.
+
+      ! Parameters of standard orbital system (tabularized intercalation cycle)
+      ! -----------------------------------------------------------------------
+      call ESMF_ConfigGetAttribute (CF, &
+         ECC, label="ECCENTRICITY:", &
+         default=DEFAULT_ORBIT_ECCENTRICITY, _RC)
+
+      call ESMF_ConfigGetAttribute (CF, &
+         OB, label="OBLIQUITY:", &
+         default=DEFAULT_ORBIT_OBLIQUITY, _RC)
+
+      call ESMF_ConfigGetAttribute (CF, &
+         PER, label="PERIHELION:", &
+         default=DEFAULT_ORBIT_PERIHELION, _RC)
+
+      call ESMF_ConfigGetAttribute (CF, &
+         EQNX, label="EQUINOX:", &
+         default=DEFAULT_ORBIT_EQUINOX, _RC)
```

Comment:
I need to understand why the `MAPL_GetResource` procedure did not work.   we may have reasons that we want to intercept these calls in the future.